### ISSUE TITLE
Pretrained word embeddings yield lower scores

### ISSUE THREAD

**Issue by henrik2706** [ufal/udpipe#16]
First of all, thanks a lot for making this tool available. I am using it in my MSc thesis on parsing of Norwegian language using neural net parsers and word embeddings.

In my experiments I have first run the parser on my dependency treebank without using pretrained word embeddings, in order to establish a baseline. I have then trained word2vec embeddings using the Norwegian Newspaper Corpus with the parameters specified in gen.sh. In other parsers, such as [Dyer et. al.'s LSTM Parser](https://github.com/clab/lstm-parser), this yields a significant increase in both LAS and UAS. In UDPipe however, my results drop by approximately .5 LAS and .1 UAS. I find this rather curious.

Another thing I find odd, is that UDPipe reports slightly lower scores when run in accuracy mode, than the scores reported by the CoNLL Shared Task's eval.pl script.

The parameters I use are
`./udpipe --train --parser=embedding_form_file=embedding.vec --parser=embedding_form=50 model.output training.conll`
`./udpipe --parse --input=conllu --output=conllu --outfile=result.conll model.output dev.conll`
`./udpipe --accuracy --parse model.output dev.conll`
`perl eval.pl -s result.conll -g dev.conll > eval_result.txt`

I suspect that I am doing something wrong, but I simply cannot figure out what. I would be grateful if you could point me in the right direction.

**Comment by foxik** [comment]
Yes, it is unexpected that precomputed word embeddings do not help -- even precomputing the embeddings on UD itself helps compared to no precomputation of word embeddings; I am not sure what went wrong.

As for the differences in UDPipe and `eval.pl`: which `eval.pl` script are you exactly using? Come CoNLL shared task parsing scripts (CoNLL 2006, if I recall correctly) deliberately ignored punctuation.

**Comment by foxik** [comment]
As for the accuracy scores, there was an error in UDPipe accuracy reporting -- that now works as expected (as close to the official evaluation script as possible).
